### PR TITLE
Invoice fixups

### LINKED ISSUE #759: Regression: clients unable to accept invitations [Epic2]
## Issue - Bug

image

Steps to reproduce:

1. Go to Team Settings, invite new team member
2. As new team member, click on invitation link
3. Observe ducks forever

I'm surprised that this wasn't tripped by our unit tests!

- **OS:** OSX
- **Browser:** Chrome

### LINKED ISSUE #792: invoice final touches
@ackernaut got a couple final touches on the invoices for ya:
screen shot 2017-03-09 at 6 25 18 pm

this looks a little weird since we don't have any adjustments for last month. should we just nuke that section if it's empty?

screen shot 2017-03-09 at 6 26 33 pm
certain things about this list i don't like:
- an upcoming invoice while i'm trialing gives me the heebie jeebies. any good ideas on how we can eloquently say "hey, this is what it'll cost you after your trial runs out, if you wanna keep using us. if not, then you don't pay anything".
- an invoice for $0.00 with a "paid on" line seems weird, can we wordsmith?
- can we make it more obvious that the first invoice is a demo? should we just hide it? i like having _something_ there, but i could go either way.

@tayamueller get in here with those comments

### LINKED ISSUE #793: No such customer exception [Epic2]
## Issue - Bug

Reproduction steps:

1. Impersonate jordan.husney@google.com user`google-oauth2|112540584686400405659`
1. Navigate to User Hub->Organizations->Jordan's Org->Settings and Billing
1. Observe invoice list does not load

Exception:

```
2017-03-10T03:11:53.559765877Z app[web.1]: DEBUG GraphQL Error: [ { Error: No such customer: cus_AEdDLtDunq0MQ5
2017-03-10T03:11:53.559870355Z app[web.1]:       at Constructor._Error (/app/node_modules/stripe/lib/Error.js:12:17)
2017-03-10T03:11:53.559877714Z app[web.1]:       at Constructor (/app/node_modules/stripe/lib/utils.js:97:13)
2017-03-10T03:11:53.559882012Z app[web.1]:       at Constructor (/app/node_modules/stripe/lib/utils.js:97:13)
2017-03-10T03:11:53.559886135Z app[web.1]:       at Function.StripeError.generate (/app/node_modules/stripe/lib/Error.js:56:14)
2017-03-10T03:11:53.559890377Z app[web.1]:       at IncomingMessage.<anonymous> (/app/node_modules/stripe/lib/StripeResource.js:137:39)
2017-03-10T03:11:53.559899592Z app[web.1]:       at emitNone (events.js:91:20)
2017-03-10T03:11:53.559903902Z app[web.1]:       at IncomingMessage.emit (events.js:186:7)
2017-03-10T03:11:53.559907950Z app[web.1]:       at IncomingMessage.EventEmitter.emit (/app/node_modules/sc-domain/index.js:12:31)
2017-03-10T03:11:53.559912407Z app[web.1]:       at endReadableNT (_stream_readable.js:974:12)
2017-03-10T03:11:53.559916529Z app[web.1]:       at _combinedTickCallback (internal/process/next_tick.js:74:11)
2017-03-10T03:11:53.559920750Z app[web.1]:       at process._tickDomainCallback (internal/process/next_tick.js:122:9)
2017-03-10T03:11:53.559924897Z app[web.1]:     message: 'No such customer: cus_AEdDLtDunq0MQ5',
2017-03-10T03:11:53.559929150Z app[web.1]:     locations: [ [Object] ],
2017-03-10T03:11:53.559933182Z app[web.1]:     path: [ 'invoiceList' ] } ]
```

### PR DESCRIPTION
**Pull request by mattkrick**
began writing webhook tests for stripe (ran out of time)

- [ ] fix #759
- [ ] fix #793 
- [ ] fix #792 
  - [ ] last month's adjustments section is hidden if possible on invoice
  - [ ] total/previous balance is hidden if possible on invoice
  - [ ] invoice for trial is hidden on the OrgBilling page
  - [ ] upcoming invoice on OrgBilling page reactively updates when the total goes invalid (add/remove user, pause/unpause user, they add a credit card & extend the trial by a month, we manually go into stripe & set something like giving them a free month)

### CODE REVIEW

File: src/server/__tests__/teardown/teardown.js [hunk 1]
```diff
@@ -16,5 +16,6 @@ export default async function teardown(results) {
     }
     await r.getPoolMaster().drain();
   }
+  await new Promise(res => setTimeout(() => res(), 50));
```
Comment (jordanh) [inline src/server/__tests__/teardown/teardown.js]: What's this for?

File: src/server/billing/stripeWebhookHandler.js [hunk 2]
```diff
@@ -5,26 +5,28 @@ import invoiceCreated from './handlers/invoiceCreated';
...
-  const event = req.body;
-  const {type} = event;
-  const objectId = event.data.object.id;
+export default (exchange) => async function stripeWebhookHandler(req, res) {
+  // code defensively here because anyone can call this endpoint
+  const event = req.body || {};
+  const {data = {}, type} = event;
+  const dataObject = data.object || {};
+  const objectId = dataObject.id;
   console.log('webhook received', type, objectId);
```
Comment (jordanh) [inline src/server/billing/stripeWebhookHandler.js]: You wanna keep these console messages? (maybe good for starters...)

**Comment by jordanh** [comment]
Lookin' real close here! I see a failure in the unit test `src/server/billing/handlers/__tests__/customerSourceUpdated.test.js`.

I'll be here off-and-on throughout the afternoon. What's your plan, hoy?

File: src/server/__tests__/teardown/teardown.js (hunk 1 above)
Comment (mattkrick) [inline src/server/__tests__/teardown/teardown.js]: jest does a crummy job of flushing to stdout before exiting. if i give it an extra 50ms, then all my console.log statements come through

File: src/server/billing/stripeWebhookHandler.js (hunk 2 above)
Comment (mattkrick) [inline src/server/billing/stripeWebhookHandler.js]: i could go either way

**Comment by mattkrick** [comment]
i think we're real close, just wanna fix up that failing test. it doesn't fail locally, so it's a bit of a mystery right now

**Comment by mattkrick** [comment]
i'm lacking some tests, but there no reason not to merge. @jordanh hit up the checklist & lemme know how it looks!

**Pull request merged**